client/data/help/use-help-search-query.ts

```diff
@@ -1,26 +1,90 @@
-import { useLocale } from '@automattic/i18n-utils';
-import { useQuery } from '@tanstack/react-query';
-import wpcomRequest from 'wpcom-proxy-request';
-import type { LinksForSection } from '@automattic/data-stores';
+import { useQueryClient, useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+
+export interface SearchResult {
+	link: string;
+	title: string | React.ReactChild;
+	content?: string;
+	icon?: string;
+	post_id?: number;
+	blog_id?: number;
+	source?: string;
+}
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
 export const useHelpSearchQuery = (
 	search: string,
-	queryOptions: Record< string, unknown > = {}
+	locale = 'en',
+	queryOptions: Record< string, unknown > = {},
+	sectionName = ''
 ) => {
-	const locale = useLocale();
-	const params = new URLSearchParams( {
-		include_post_id: '1',
-		locale,
-	} );
+	const queryClient = useQueryClient();
 
-	if ( search ) {
-		params.append( 'query', search );
+	// set up article list fetching
+	let fetchArticlesAPI: () => Promise< unknown >;
+	const params = `?query=${ encodeURIComponent( search ) }&locale=${ encodeURIComponent(
+		locale
+	) }&section=${ encodeURIComponent( sectionName ) }`;
+	if ( canAccessWpcomApis() ) {
+		fetchArticlesAPI = () =>
+			wpcomRequest( {
+				path: `help/search/wpcom${ params }`,
+				apiNamespace: 'wpcom/v2/',
+				apiVersion: '2',
+			} );
+	} else {
+		fetchArticlesAPI = () =>
+			apiFetch( {
+				global: true,
+				path: `/help-center/search${ params }`,
+			} as APIFetchOptions );
 	}
 
-	return useQuery< { wordpress_support_links: LinksForSection[] } >( {
-		queryKey: [ 'help', search ],
-		queryFn: () =>
-			wpcomRequest( { path: '/help/search', query: params.toString(), apiVersion: '1.1' } ),
-		enabled: !! search,
+	// set up article fetching
+	let fetchArticleAPI: ( blogId: number, postId: number ) => Promise< { content: string } >;
+	if ( canAccessWpcomApis() ) {
+		fetchArticleAPI = ( blogId: number, postId: number ) =>
+			wpcomRequest( {
+				path: `help/article/${ blogId }/${ postId }`,
+				apiNamespace: 'wpcom/v2/',
+				apiVersion: '2',
+			} );
+	} else {
+		fetchArticleAPI = ( blogId: number, postId: number ) =>
+			apiFetch( {
+				global: true,
+				path: `/help-center/fetch-post?blog_id=${ blogId }&post_id=${ postId }`,
+			} as APIFetchOptions );
+	}
+
+	return useQuery< any >( {
+		queryKey: [ 'help-center-search', search, sectionName ],
+		queryFn: fetchArticlesAPI,
+		onSuccess: async ( data ) => {
```

Comment:
Please take into consideration that `onSuccess` callbacks for queries (not mutations!) are [deprecated](https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#callbacks-on-usequery-and-queryobserver-have-been-removed) and will be removed in the next major version of `@tanstack/query` so it may be worthwhile to explore different solutions to fetching additional content.